workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/OverlayDialogHolder.kt

```diff
@@ -0,0 +1,93 @@
+package com.squareup.workflow1.ui.container
+
+import android.app.Dialog
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewEnvironmentKey
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.compatible
+import com.squareup.workflow1.ui.container.OverlayDialogHolder.Companion.InOverlay
+import com.squareup.workflow1.ui.container.OverlayDialogHolder.Companion.NoOverlay
+import com.squareup.workflow1.ui.show
+
+/**
+ * Associates a [dialog] with a function ([runner]) that can update it to display instances
+ * of [OverlayT]. Also holds a reference to the [ViewEnvironment][environment] that was
+ * most recently used to update the [dialog].
+ */
+@WorkflowUiExperimentalApi
+public interface OverlayDialogHolder<in OverlayT : Overlay> {
+  /** The [Dialog] managed by this holder, and updated via [runner] */
+  public val dialog: Dialog
+
+  /** The [ViewEnvironment] that was provided the last time [dialog] was updated by [runner]. */
+  public val environment: ViewEnvironment
+
+  /**
+   * The function that is run by [show] to update [dialog] with a new [Screen] rendering and
+   * [ViewEnvironment].
+   */
+  public val runner: (rendering: OverlayT, environment: ViewEnvironment) -> Unit
```

Comment:
I thought it'd be the runner factory at first glance. Lol.
Why not update?
(no biggies)